### PR TITLE
Fix: Voting, viewing an image or visiting a URL should set a post as read

### LINKED ISSUE #336: Feature: Voting or viewing post image from the main feed should mark post as read
Basically title, currently when in the main feed if a user upvotes/downvotes or click on a post image, the post isn't marked as read. 
Changing the behavior to mark a post as read on these interactions in combination with "hide read posts" makes it very simple to quickly go through a feed and mark all as read.

I believe Jerboa already does this for votes, but not for image click.

### PR DESCRIPTION
**Pull request by mdecourcy**
Resolves #336

### CODE REVIEW

File: src/features/post/postSlice.ts [hunk 1]
```diff
@@ -194,9 +194,10 @@ export const voteOnPost =
         score: vote,
         auth: jwt,
       });
+      dispatch(setPostRead(postId));
```
Comment (aeharding) [inline src/features/post/postSlice.ts:197]: please move this into finally block

File: src/features/post/shared/Embed.tsx [hunk 2]
```diff
@@ -71,14 +73,20 @@ interface EmbedProps {
 
 export default function Embed({ post, className }: EmbedProps) {
   const [error, setError] = useState(false);
+  const dispatch = useAppDispatch();
+
+  const handleLinkClick = (e: { stopPropagation: () => void }) => {
```
Comment (aeharding) [inline src/features/post/shared/Embed.tsx:78]: `e: MouseEvent` (imported from React)

File: src/features/gallery/GalleryPostActions.tsx [hunk 3]
```diff
@@ -60,6 +62,7 @@ export default function GalleryPostActions({ post }: GalleryPostActionsProps) {
       </Section>
       <div
         onClick={() => {
+          dispatch(setPostRead(post.post.id));
```
Comment (aeharding) [inline src/features/gallery/GalleryPostActions.tsx:65]: This should not be needed. Post marked as read when you open the detail page

File: src/features/gallery/GalleryProvider.tsx [hunk 4]
```diff
@@ -86,8 +89,10 @@ export default function GalleryProvider({ children }: GalleryProviderProps) {
       animationType?: PreparedPhotoSwipeOptions["showHideAnimationType"]
     ) => {
       if (lightboxRef.current) return;
+      if (!post) return;
```
Comment (aeharding) [inline src/features/gallery/GalleryProvider.tsx:92]: Please don't guard here, some non post components may use this. Instead, `if(post) { do the dispatching }`

**Review by aeharding — COMMENTED** [review]
Thanks for the PR, just a bit of feedback

File: src/features/post/postSlice.ts (hunk 1 above)
Comment (mdecourcy) [reply src/features/post/postSlice.ts:197]: Will do

File: src/features/gallery/GalleryPostActions.tsx (hunk 3 above)
Comment (mdecourcy) [reply src/features/gallery/GalleryPostActions.tsx:65]: My bad, didn't ctrl + s this file after removing

**Comment by mdecourcy** [comment]
New commits should resolve @aeharding

File: src/features/post/postSlice.ts [hunk 5]
```diff
@@ -196,8 +196,10 @@ export const voteOnPost =
       });
     } catch (error) {
       dispatch(updatePostVote({ postId, vote: oldVote }));
-
+      dispatch(setPostRead(postId));
```
Comment (aeharding) [inline src/features/post/postSlice.ts:199]: this can be removed too now that it's in the finally

**Review by aeharding — APPROVED** [review]
Nice job! Thanks for the quick adjustments

**Pull request merged**